src/presenter/event-presenter.js

```diff
@@ -0,0 +1,75 @@
+import WayPoint from '../view/way-point.js';
+import EditingForm from '../view/editing-form.js';
+import { render, replace} from '../framework/render.js';
+import { updateItem } from '../util.js';
+import { Mode } from '../const.js';
+export default class EventPresenter {
+  #event = null;
+  #destinations = [];
+  #offers = [];
+  #eventListContainer = null;
+  #handlePointUpdate = null;
+  #tripEventView = null;
+  #eventEditView = null;
+  #mode = Mode.DEFAULT;
+  constructor({eventListContainer, destinations,offers,onPointUpdate}) {
+    this.#destinations = destinations;
+    this.#offers = offers;
+    this.#eventListContainer = eventListContainer;
+    this.#handlePointUpdate = onPointUpdate;
+  }
+
+  init(event) {
+    this.#event = event;
+    this.#renderTripEvent(this.#event, this.#destinations, this.#offers);
+  }
+
+  resetView() {
+    if(this.#mode === Mode.EDIT) {
+      this.#switchToViewMode();
+    }
+  }
+
+  #renderTripEvent(event, destinations, offers) {
+    const prevEventView = this.#tripEventView;
+
+    this.#tripEventView = new WayPoint({event,destinations, offers, onClick: () => {
+      this.#switchToEditMode();
+    }, onFavouriteBtnClick: () => {
+      const updatePoint = updateItem(event, {isFavorite: !event.isFavorite});
+      this.#handlePointUpdate(updatePoint);
+    }});
+
+    this.#eventEditView = new EditingForm({event,destinations, offers, onSubmit: () => {
+      this.#switchToViewMode();
+    }, onClick:this.#switchToViewMode()});
```

Comment:
Проблема в этой строке. Здесь ты в `onClick` передаёшь не колбэк, который выполнит что-то, а результат вызова функции `this.#switchToViewMode()`, а т.к. в самом начале у тебя ещё нет нужных данных для работы метода, у тебя `replace()` и падает с ошибкой. Тут должно быть вот так:
```suggestion
    }, onClick: () => {this.#switchToViewMode()}});
```